caas/kubernetes/provider/k8s.go

```diff
@@ -187,6 +191,14 @@ func newK8sBroker(
 		return nil, errors.NotValidf("modelUUID is required")
 	}
 
+	isLegacy, err := utils.IsLegacyModelLabels(
+		newCfg.Config.Name(), k8sClient.CoreV1().Namespaces())
+	if err != nil {
+		logger.Warningf("determining legacy label status for model %s: %v",
+			newCfg.Config.Name(), err)
+		isLegacy = true
```

Comment:
It used to but there was a test case that cause it to break. Let me take another look.

Reply:
Yeah I'd like to see this handled. Considering this changes how we update the Kubernetes state, it could be very dangerous if we operated on a namespace without first understanding it.